planning/autoware_planning_validator/src/utils.cpp

```diff
@@ -146,45 +146,99 @@ std::pair<double, size_t> calcMaxCurvature(const Trajectory & trajectory)
     return {0.0, 0};
   }
 
-  std::vector<double> curvature_arr;
-  calcCurvature(trajectory, curvature_arr);
+  std::vector<double> curvature_vector;
+  calcCurvature(trajectory, curvature_vector);
 
-  const auto max_curvature_it = std::max_element(curvature_arr.begin(), curvature_arr.end());
-  const size_t index = std::distance(curvature_arr.begin(), max_curvature_it);
+  const auto max_curvature_it = std::max_element(curvature_vector.begin(), curvature_vector.end());
+  const size_t index = std::distance(curvature_vector.begin(), max_curvature_it);
 
   return {*max_curvature_it, index};
 }
 
+void calc_interval_distance(
+  const Trajectory & trajectory, std::vector<double> & interval_distance_vector)
+{
+  interval_distance_vector.clear();
+
+  if (trajectory.points.size() <= 1) {
+    return;
+  }
+
+  interval_distance_vector.resize(trajectory.points.size() - 1, 0.0);
+
+  for (size_t i = 0; i < trajectory.points.size() - 1; ++i) {
+    const auto d = calc_distance2d(trajectory.points.at(i), trajectory.points.at(i + 1));
+    interval_distance_vector.at(i) = d;
+  }
+}
+
 std::pair<double, size_t> calcMaxIntervalDistance(const Trajectory & trajectory)
 {
-  if (trajectory.points.size() < 2) {
+  if (trajectory.points.size() <= 1) {
     return {0.0, 0};
   }
+  std::vector<double> interval_distance_vector;
+  calc_interval_distance(trajectory, interval_distance_vector);
 
-  double max_interval_distances = 0.0;
-  size_t max_index = 0;
-  for (size_t i = 1; i < trajectory.points.size(); ++i) {
-    const auto d = calc_distance2d(trajectory.points.at(i), trajectory.points.at(i - 1));
-    if (max_interval_distances < std::abs(d)) {
-      takeBigger(max_interval_distances, max_index, std::abs(d), i);
-    }
+  if (interval_distance_vector.empty()) {
+    return {0.0, 0};
+  }
+
+  const auto max_interval_it =
+    std::max_element(interval_distance_vector.begin(), interval_distance_vector.end());
+  const size_t max_index = std::distance(interval_distance_vector.begin(), max_interval_it);
+
+  return {*max_interval_it, max_index};
+}
+
+void calc_lateral_acceleration(
+  const Trajectory & trajectory, std::vector<double> & lateral_acceleration_vector)
+{
+  lateral_acceleration_vector.clear();
+
+  // We need at least three points to compute curvature
+  if (trajectory.points.size() < 3) {
+    lateral_acceleration_vector.resize(trajectory.points.size(), 0.0);
```

Comment:
we also can move `lateral_acceleration_vector.resize(trajectory.points.size(), 0.0);` to outside of {} scope here too

Reply:
Thanks! addressed in https://github.com/autowarefoundation/autoware_universe/pull/10385/commits/69a27ce5729676890081aa34711f3a6f515c3f9a